poetry/installation/executor.py

```diff
@@ -759,6 +759,10 @@ def _save_url_reference(self, operation: "OperationTypes") -> None:
                     encoding="utf-8",
                 )
 
+                if dist._path.joinpath("RECORD").exists():
+                    with dist._path.joinpath("RECORD").open(mode="a") as record:
+                        record.write(str(dist._path.joinpath("direct_url.json")))
```

Comment:
I don't think it will work since the `RECORD` file is a CSV file so the added line should be a valid CSV line.

Also, it should not be an absolute path to the file but relative to the site-packages directory.

```suggestion
                record = dist._path.joinpath("RECORD")
                if record.exists():
                    with record.open(mode="a") as f:
                        writer = csv.writer(f)
                        writer.writerow(
                            [str(dist._path.joinpath("direct_url.json").relative_to(record.parent.parent)), "", ""]
                        )
```

Finally, could you add a test covering this change?

Reply:
@sdispater it did work, I tried it myself. But you are right that it should be treated as CSV.

I'll look into how I can add a test. I guess I should write one which installs a package from path, then removes it - and assert that there's no lingering `direct_url.json` file in the package's dist-info folder?

Reply:
@fredrikaverpil For the test, simply ensuring that the RECORD file has the `direct_url.json` reference should be enough.

Reply:
@sdispater I've now applied your code suggestion and added an additional assert to the existing `verify_installed_distribution` test.

I'm noticing that in your code suggestion you changed the absolute path into a relative one. According to [the docs](https://packaging.python.org/specifications/recording-installed-packages/#the-record-file), it's okay to use _either_ absolute or relative. So relative seems nicer 👍 But, I just want you to know that the `poetry add ../b --editable` command resulted in the following `RECORD` on my machine (Ubuntu 20.04), where the `direct_url.json` path was the _only_ one relative:

```csv
/home/fredrik/code/repos/poetry/.venv/lib/python3.8/site-packages/b.pth,sha256=PB1gW2CKFckiIksXEn9ySceS8mhVpDf9NNMba5UTaAE,31
/home/fredrik/code/repos/poetry/.venv/bin/bclixq,sha256=oUuzygRt7DgCtrAZlWpxIDam0n76RCpGe6FU5P-zx3s,115
/home/fredrik/code/repos/poetry/.venv/lib/python3.8/site-packages/b-0.0.0.dist-info/METADATA,sha256=zaDEAXqbNAXJXxcfQbscYIbhAxFGCpoYI37JYOJNyo4,564
/home/fredrik/code/repos/poetry/.venv/lib/python3.8/site-packages/b-0.0.0.dist-info/INSTALLER,sha256=gcMg69IJ3d31kOPdejOXCeJ_7u2OVE9WaZ5A1idCOi0,6
/home/fredrik/code/repos/poetry/.venv/lib/python3.8/site-packages/b-0.0.0.dist-info/entry_points.txt,sha256=qroUuuKxidhz-9dsg5bVbzH_PX6YmETz_ZmpKlajQ08,39
/home/fredrik/code/repos/poetry/.venv/lib/python3.8/site-packages/b-0.0.0.dist-info/RECORD,,
b-0.0.0.dist-info/direct_url.json,,
```